File: ruoze-spark/ruoze-spark-sb/src/main/java/com/ruozedata/ruozesparksb/controller/AccessController.java
package com.ruozedata.ruozesparksb.controller;

import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * https://www.imooc.com/article/295920
 */
@RestController
@RequestMapping(value = "/access")
public class AccessController {

    @RequestMapping(value = "/{date}")
    public String processAccess(@PathVariable String date) {
        System.out.println(date);
        return date;
    }
}

File: ruoze-spark/ruoze-spark-sb/src/main/scala/com/ruozedata/ruozesparksb/controller/AccessController.scala
package com.ruozedata.ruozesparksb.controller

import com.ruozedata.ruozesparksb.service.AccessService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.{PathVariable, RequestMapping, RestController}

@RestController
@RequestMapping(value = Array("/access"))
class AccessController @Autowired()(val accessService: AccessService) {

  // http://hadoop:9527/spark/access/20190101
  @RequestMapping(value = Array("/{date}"))
  def processAccessLog(@PathVariable date: String): Unit = {
    accessService.processAccessLog(date)
  }

}
